app/post_metadata.go

```diff
@@ -102,11 +102,14 @@ func (a *App) PreparePostForClient(originalPost *model.Post, isNewPost bool, isE
 
 	a.OverrideIconURLIfEmoji(post)
 
-	post.Metadata = &model.PostMetadata{}
+	if post.Metadata == nil {
+		post.Metadata = &model.PostMetadata{}
+	}
```

Comment:
What's this change for? Are we calling this method twice on a single post, or are we providing metadata from somewhere else?

I'd be worried about changing this since it means that calling this method twice on a single post won't result in the same object. If you do that, it'll have a a doubled embeds array